main.go

```diff
@@ -56,6 +63,20 @@ func realMain() int {
 		}
 	}()
 
+	// Create the version checker
+	checker, err := versioncheck.New(io, versionCheckStatePath)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "failed to configure version checker: %v\n", err)
+		return 1
+	}
+
+	// Start checking for a new version as soon as possible
+	go func() {
```

Comment:
Since we plan to only show the new version prompt once per 24 hours, is there a need to put this on a go routine?

Reply:
The reason it is in a goroutine is so that the command the user wants to run and the version checking can run in parallel. This means the user ideally doesn't experience any additional delay because we are doing a version check.

Reply:
My concern was more on the lines that a user hits the end while go routine isn't finished. So prompt for version check never happens as it is a slow path.

Reply:
That is the intention. If the command finishes before the version check finishes, you just cancel the version check.